partner_invoicing_mode/models/sale_order.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2020 Camptocamp SA
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+from odoo import api, fields, models
+from odoo.fields import Datetime
+from odoo.osv.expression import AND
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    invoicing_mode = fields.Selection(
+        related="partner_invoice_id.invoicing_mode",
+        store=True,
+        index=True,
+    )
+
+    one_invoice_per_order = fields.Boolean(
+        compute="_compute_one_invoice_per_order",
+        readonly=False,
+        store=True,
+        help="You can check or uncheck this if you want the periodic invoicing"
+        " grouping this sale order with other ones or not.",
+    )
+
+    @api.depends("partner_invoice_id")
+    def _compute_one_invoice_per_order(self):
+        # We depends only on partner as if we change it, we should recompute
+        # but not if the parameter on it has changed (this allows to set a different
+        # value on each sale order).
+        for order in self:
+            order.one_invoice_per_order = order.partner_invoice_id.one_invoice_per_order
+
+    @api.model
+    def cron_generate_standard_invoices(self):
+        company_ids = self._get_companies_standard_invoicing()
+        if company_ids:
+            self.generate_invoices(company_ids)
+
+    @api.model
+    def _get_generate_invoices_state_domain(self):
+        return [("invoice_status", "=", "to invoice")]
+
+    @api.model
+    def _get_generate_invoices_domain(self, companies, invoicing_mode="standard"):
+        return AND(
+            [
+                self._get_generate_invoices_state_domain(),
+                [
+                    ("invoicing_mode", "=", invoicing_mode),
+                    ("company_id", "in", companies.ids),
+                ],
+            ]
+        )
+
+    @api.model
+    def generate_invoices(
+        self,
+        companies=None,
+        invoicing_mode="standard",
+        last_execution_field="invoicing_mode_standard_last_execution",
+    ):
+        """
+        Generate invoices in job queues depending on the invoicing
+        mode (stadndard by default)
+        """
+        if companies is None:
+            companies = self.env.company
+        domain = self._get_generate_invoices_domain(
+            companies=companies, invoicing_mode=invoicing_mode
+        )
+        saleorder_groups = self.read_group(
+            domain,
+            ["partner_invoice_id", "sale_ids:array_agg(id)"],
+            groupby=self._get_invoice_grouping_keys(),
+            lazy=False,
+        )
+        for saleorder_group in saleorder_groups:
+            self.with_delay()._generate_invoices_by_partner(saleorder_group["sale_ids"])
```

Comment:
[IMP] Can you fill the `description` and the `identity_key` to avoid duplicates jobs for the same records.